components/Author.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+
+import AuthorCollection from '../interfaces/AuthorCollection'
+import { MarkdownFileWithUrl } from '../pages/authors'
+
+import ContentCardList from './ContentCardList'
+import PageLayout from './PageLayout'
+
+export interface Props {
+ author?: AuthorCollection
+ records: MarkdownFileWithUrl[]
+}
```

Comment:
Suggestion to make author non-optional in the props. 
```suggestion
export interface Props {
 author: AuthorCollection
 records: MarkdownFileWithUrl[]
}
```

Reply:
One use case that comes to mind for having an optional author is if we want a generic "Sourcegraph" authored post, which I think is strange to write in as an author.

Reply:
I guess this means we can keep author optional, since we can have generic "Sourcegraph" authored articles too.

Reply:
After thinking about this, I think this is valuable feedback and is valid. An article can still be shown if it has no author but  when the author page is being accessed with an invalid author, the author page should not show if there is no author. I have updated the PR with this feedback.

Reply:
@Princesso Yes, that's what I meant with my suggestion: to make author non-optional in the `Props` interface of this author page component. @ltagliaferri To clarify, we can still have an optional `author` field in the front-matter in markdown files, this suggestion applies only to the author page component's props.